wasmtest.sh

```diff
@@ -0,0 +1,256 @@
+# To run all files use: ./wasmtest.sh allrun
+# To test all tests from the file user give use: ./wasmtest.sh files
+# To compile all tests from the file user give use: ./wasmtest.sh filescompile
+# To run all tests from the file user give use: ./wasmtest.sh filesrun
+
+glibc_base="/home/lind-wasm/src/glibc"
+wasmtime_base="/home/lind-wasm/src/wasmtime"
+
+CC="${CLANG:=/home/lind-wasm/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04}/bin/clang"
+
+export_cmd="export LD_LIBRARY_PATH=$wasmtime_base/crates/rustposix:\$LD_LIBRARY_PATH"
+
+compile_test_cmd_fork_test="$CC -pthread --target=wasm32-unknown-wasi --sysroot $glibc_base/sysroot -Wl,--import-memory,--export-memory,--max-memory=67108864,--export="__stack_pointer" [input] -g -O0 -o [output] && wasm-opt --asyncify --debuginfo [output] -o [output]"
+precompile_wasm="$wasmtime_base/target/debug/wasmtime compile [input] -o [output]"
+run_cmd_precompile="$wasmtime_base/target/debug/wasmtime run --allow-precompiled --wasi threads=y --wasi preview2=n [target]"
+
+test_file_base="/home/lind-wasm/tests/unit-tests"
+
+#color codes for terminal output
+RED='\033[31m'
+GREEN='\033[32m'
+RESET='\033[0m'
+
+#function to compile a single test file
+compile_single_test() {
+    local test_name="$1"
+
+    source_file=$(find "$test_file_base" -type f -name "${test_name}" 2>/dev/null | head -n 1)
+
+    if [ -z "$source_file" ]; then
+        echo -e "${RED}Error: Source file ${test_name} not found in $test_file_base.${RESET}"
+        exit 1
+    fi
+
+    output_dir=$(dirname "$source_file")
+    without_c="${test_name%??}"
+    output_file_wasm="${output_dir}/${twithout_c}.wasm"
+    output_file_cwasm="${output_dir}/${without_c}.cwasm"
+
+    eval $export_cmd
+
+    final_cmd=$(echo "$compile_test_cmd_fork_test" | sed "s|\[input\]|$source_file|g" | sed "s|\[output\]|$output_file_wasm|g")
+    pre_compile=$(echo "$precompile_wasm" | sed "s|\[input\]|$output_file_wasm|g" | sed "s|\[output\]|$output_file_cwasm|g")
+
+    final_cmd="${final_cmd} && ${pre_compile}"
+
+    echo -e "${GREEN}Compiling test: $test_name in $output_dir${RESET}"
+    echo -e "${GREEN}$final_cmd${RESET}"
+
+    if [ "$pmode" -eq 0 ]; then
+        eval "$final_cmd"
+    fi
+}
+
+#function to run a single test file
+run_single_test() {
+    local test_name="$1"
+
+    if [[ "$test_name" == /* ]]; then
+        new_test_name="${test_name#/}"
+    else
+        new_test_name="$test_name"
+    fi
+
+    target_wasm="${output_dir}/${new_test_name}.wasm"
+    target_cwasm="${output_dir}/${new_test_name}.cwasm"
+
+    shift 1
+
+    if [ -e "$target_cwasm" ]; then
+        final_cmd=$(echo "$run_cmd_precompile" | sed "s|\[target\]|$target_cwasm|")
+        final_cmd="${final_cmd} $@"
+    else
+        final_cmd=$(echo "$run_cmd" | sed "s|\[target\]|$target_wasm|")
+        final_cmd="${final_cmd} $@"
+    fi
+
+    eval $export_cmd
+
+    echo -e "${GREEN}Running: $final_cmd${RESET}"
+
+    if timeout 10s bash -c "$final_cmd"; then
```

Comment:
It might be better to make `10s` be a configurable value. By default it is 10 second, but we can support one more optional command line option to customize this value